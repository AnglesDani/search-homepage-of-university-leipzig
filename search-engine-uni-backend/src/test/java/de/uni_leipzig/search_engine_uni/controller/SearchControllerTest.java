package de.uni_leipzig.search_engine_uni.controller;

import org.approvaltests.Approvals;
import org.junit.Before;
import org.junit.Test;

public class SearchControllerTest
{
	@Before
	public void setup()
	{
		SearchControllerMockUtils.setupHttpServlet();
	}
	
	@Test
	public void approveNullInput()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(0).search(null, null));
	}
	
	@Test
	public void approveEmptyQueryInput()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(0).search("", null));
	}
	
	@Test
	public void approveFirstPageResultWithOneMorePageAvailable()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(11).search("queryDoesNotMatter", 1));
	}
	
	@Test
	public void approveSecondPageResultWithOnePageBeforeAvailable()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(11).search("queryDoesNotMatter", 2));
	}
	
	@Test
	public void approveRequestOfFifthPageReturnsSecondPageIfOnlyElevenResultsAreAvailable()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(11).search("queryDoesNotMatter", 5));
	}
	
	@Test
	public void approveNextAndPreviousPageLinksAreCorrectForSecondPageOfThreePages()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(30).search("queryDoesNotMatter", 2));
	}
	
	@Test
	public void approveNegativePageRequestForFiftySevenAvailableResults()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(57).search("queryDoesNotMatter", -1));
	}
	
	@Test
	public void approveLastPageRequestForFiftySevenAvailableResults()
	{
		Approvals.verify(SearchControllerMockUtils.createSearchController(57).search("queryDoesNotMatter", 6));
	}
}
