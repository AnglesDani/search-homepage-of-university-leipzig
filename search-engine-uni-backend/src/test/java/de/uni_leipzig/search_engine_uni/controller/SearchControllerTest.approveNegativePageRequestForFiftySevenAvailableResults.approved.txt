SearchResultPage(originalQuery=queryDoesNotMatter, page=1, resultsOnPage=[SearchResult(title=0, snippet=null, targetUrl=<http://www.example.com/url?documentId=0>;rel="targetUrl"), SearchResult(title=1, snippet=null, targetUrl=<http://www.example.com/url?documentId=1>;rel="targetUrl"), SearchResult(title=2, snippet=null, targetUrl=<http://www.example.com/url?documentId=2>;rel="targetUrl"), SearchResult(title=3, snippet=null, targetUrl=<http://www.example.com/url?documentId=3>;rel="targetUrl"), SearchResult(title=4, snippet=null, targetUrl=<http://www.example.com/url?documentId=4>;rel="targetUrl"), SearchResult(title=5, snippet=null, targetUrl=<http://www.example.com/url?documentId=5>;rel="targetUrl"), SearchResult(title=6, snippet=null, targetUrl=<http://www.example.com/url?documentId=6>;rel="targetUrl"), SearchResult(title=7, snippet=null, targetUrl=<http://www.example.com/url?documentId=7>;rel="targetUrl"), SearchResult(title=8, snippet=null, targetUrl=<http://www.example.com/url?documentId=8>;rel="targetUrl"), SearchResult(title=9, snippet=null, targetUrl=<http://www.example.com/url?documentId=9>;rel="targetUrl")], totalHits=57, nextPage=<http://www.example.com/?originalQuery=queryDoesNotMatter&currentPage=2>;rel="next", previousPage=null, firstPageLink=null, lastPageLink=<http://www.example.com/?originalQuery=queryDoesNotMatter&currentPage=6>;rel="6", namedPageLinksBefore=[], namedPageLinksAfter=[<http://www.example.com/?originalQuery=queryDoesNotMatter&currentPage=2>;rel="2", <http://www.example.com/?originalQuery=queryDoesNotMatter&currentPage=3>;rel="3", <http://www.example.com/?originalQuery=queryDoesNotMatter&currentPage=4>;rel="4"])